### PR TITLE
fix(workflow): remove and fix ordering keys

### PR DESCRIPTION
**Pull request by Yelinz**
remove non exsisting ordering keys for work item and
task, fix relational ordering keys for case and work item

removes allow_all_forms and is_published from work item and task
rename is_archived on work item to task__is_archived
append workflow__ to keys on case which are from workflow relation

### CODE REVIEW

File: caluma/caluma_workflow/filters.py [hunk 1]
```diff
@@ -109,13 +109,18 @@ class CaseOrderSet(BaseFilterSet):
...
+            "workflow__name",
+            "workflow__description",
+            "workflow__is_archived",
+            "workflow__is_published",
+            "workflow__allow_all_forms",
+            "document__form__slug",
+            "document__form__name",
+            "document__form__description",
+            "document__form__is_archived",
+            "document__form__is_published",
```
Comment (winged) [inline caluma/caluma_workflow/filters.py:123]: I'm not sure those are all useful. It's only about ordering after all. For example, are you sure we will (soon-ish) need to sort cases by the fact whether it's document's form is actually published? Or the form's description? I don't think so. 

Therefore please only add new order-by fields that are either immediately used or easily conceivable of a use case. Most of those are not, IMHO

File: caluma/caluma_workflow/filters.py [hunk 2]
```diff
@@ -200,14 +203,12 @@ class WorkItemOrderSet(BaseFilterSet):
...
             "closed_at",
-            "allow_all_forms",
             "description",
-            "is_archived",
-            "is_published",
             "name",
             "deadline",
             "status",
             "slug",
+            "task__is_archived",
```
Comment (winged) [inline caluma/caluma_workflow/filters.py:211]: Same here - do we really need to sort work items by whether the associated task is archived? This is not something the users of work items would be interested in

**Review by winged — CHANGES_REQUESTED** [review]

File: caluma/caluma_workflow/filters.py (hunk 2 above)
Comment (Yelinz) [reply caluma/caluma_workflow/filters.py]: removed

File: caluma/caluma_workflow/filters.py (hunk 1 above)
Comment (Yelinz) [reply caluma/caluma_workflow/filters.py]: only kept slug and name

Comment (winged) [reply caluma/caluma_workflow/filters.py]: Still not sure we really need those

**Review by winged — APPROVED** [review]

**Pull request merged**